OpenRA.Game/Graphics/Util.cs

```diff
@@ -191,6 +223,69 @@ public static void FastCopyIntoSprite(Sprite dest, Png src)
 			}
 		}
 
+		/// <summary>Rotates a quad (rectangle) rotated along its Z-axis</summary>
+		/// <param name="tl">The top left vertex of the quad</param>
+		/// <param name="size">A float3 containing the X, Y, and Z lengths of the quad</param>
+		/// <param name="rotation">The number of radians to rotate by</param>
+		/// <returns>An array of four vertices representing the rotated quad (top-left, top-right, bottom-right, bottom-left)</returns>
+		public static float3[] RotateQuad(float3 tl, float3 size, float rotation)
+		{
+			var center = tl + 0.5f * size;
+			var angleSin = (float)Math.Sin(-rotation);
+			var angleCos = (float)Math.Cos(-rotation);
+
+			// Rotated offset for +/- x with +/- y
+			var ra = 0.5f * new float3(
+				size.X * angleCos - size.Y * angleSin,
+				size.X * angleSin + size.Y * angleCos,
+				(size.X * angleSin + size.Y * angleCos) * size.Z / size.Y);
+
+			// Rotated offset for +/- x with -/+ y
+			var rb = 0.5f * new float3(
+				size.X * angleCos + size.Y * angleSin,
+				size.X * angleSin - size.Y * angleCos,
+				(size.X * angleSin - size.Y * angleCos) * size.Z / size.Y);
+
+			return new float3[]
+			{
+				center - ra,
+				center + rb,
+				center + ra,
+				center - rb
+			};
+		}
+
+		/// <summary>
+		/// Returns the bounds of an object. Used for determining which objects need to be rendered on screen, and which do not.
+		/// </summary>
+		/// <param name="offset">The top left vertex of the object</param>
+		/// <param name="size">A float 3 containing the X, Y, and Z lengths of the object</param>
+		/// <param name="rotation">The angle to rotate the object by (use WAngle.Zero if there is no rotation)</param>
```

Comment:
`WAngle.Zero` is not a valid float!

Reply:
Ah, this must have been a classic case of documentation not being maintained with the code. I believe this function previously was using `WAngle` rather than floats. Have updated it now.